nemo/collections/nlp/modules/common/bert_module.py

```diff
@@ -50,7 +51,12 @@ def save_to(self, save_path: str):
 
     def restore_weights(self, restore_path: str):
         """Restores module/model's weights"""
-        logging.info(f"restore from {restore_path}")
+        logging.info(f"Restoring weights from {restore_path}")
+
+        if not os.path.exists(restore_path):
+            logging.warning(f'Path {restore_path} not found')
+            return
```

Comment:
This part is not called by core automatically, but explicitly called by user. It should fail fast if they give wrong paths

Reply:
When the model is trained from a provided bert_checkpoint path, it will store this to the config. And then during restore_from(), the file could be missing (this is to fix yesterday's error, we discussed in slack). I agree, the fast fail would be better, but then another hack to make restoration form .nemo file is needed. I can remove the missing checkpoint path from the config wit the update_config..() call, but every nlp model should do the same then. @titu1994 do you think it's better to that?

Reply:
Ok this is for restoration code path, makes sense. Theres two ways to approach this
1) "clean but heavy way" - add these checkpoints as artifacts and make the .nemo model massive in size (especially for larger BERT/ variants). This is kinda bad cause the actual NER model is just a fraction of the size.
2) "workaround way" - as shown here, it shifts burden to individual models and logs missing filepaths. If every single NLP model is going to do this, I would suggest having a baseclass for NLP (like how ASR has the unique ASRModel which is what all other models subclass from. In this NLP base class, you could do this work around.

I say this NLP Baseclass is a better approach, in the future user who wishes to build models using NeMo for NLP should have the least friction. They should not have to consider such manifest workarounds for a good user experience of saving and restoring .nemo files. 

The NLP refactor can be done in a future PR, for now this workaround is ok.